c7n/resources/kms.py

```diff
@@ -114,3 +138,35 @@ def process_key(self, key):
             Scope=key['AliasName'][6:])
 
         return key
+
+class ResourceKmsKeyAlias(ValueFilter):
+
+    schema = type_schema('kms-alias', rinherit=ValueFilter.schema)
+    def get_matching_aliases(self, resources, event=None):
+
+        key_aliases = KeyAlias(self.manager.ctx, {}).resources()
+
+        # def _user_kms_alias(resource):
+        #     kms_key_id = resource.get('KmsKeyId')
+        #     target_key_aliases = [k for k in key_aliases
+        #         if k.get('TargetKeyId') and kms_key_id
+        #         and k.get('TargetKeyId') in kms_key_id]
+        #     for kms_alias in target_key_aliases:
+        #         resource['Alias'] = kms_alias
+        #         break
+
+        # key_dict = None
+
+        # for r in resources:
+        # kms_key_id = r.get('KmsKeyId')
+        key_alias_dict = {r.get('KmsKeyId'): a for r in resources for a in key_aliases
```

Comment:
i was referencing more creating a lookup dict for key id to alias, and then using that for the resources check

ie more like this
````python
key_aliases = {a['TargetKeyId']: a['KeyAlias'] for a in key_aliases if not a['KeyAlias']}
for r in resources:
      r['KeyAlias'] = key_aliases.get(r.get('KmsKeyId'))